### PR TITLE
continuous delivery to GKE

### PR DESCRIPTION
**Pull request by 0lmi**
https://tracker.mender.io/browse/INF-1434

I checked created job and [it works](https://gitlab.com/Northern.tech/Mender/integration-test-runner/-/jobs/622732458). The only thing is missing is updating k8s deployment yaml in `sre-tools` repo. @lluiscampos can you please advice how to do this?

### CODE REVIEW

**Comment by 0lmi** [comment]
@lluiscampos looks like `publish` phase is used for CD but uploading image to docker hub is missing. Is it possible to do this with using some includes? Should not it be pushed in `.gitlab-ci-check-docker-build.yml`?
For instance in [this](https://gitlab.com/Northern.tech/Mender/integration-test-runner/-/pipelines/162791376) pipeline `build:docker` job is successful but it's no any images in [docker hub](https://hub.docker.com/r/mendersoftware/integration-test-runner/tags).

**Comment by 0lmi** [comment]
I think I got it. It has only->refs->master, so, it should be pushed only on PR merge.

**Comment by lluiscampos** [comment]
> I think I got it. It has only->refs->master, so, it should be pushed only on PR merge.

Exactly :+1:

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -14,38 +14,34 @@ include:
...
   stage: publish
   only:
     refs:
       - master
   image: debian:buster
+  before_script:
+    - export DOCKER_REPOSITORY=mendersoftware/integration-test-runner
+    - export DOCKER_TAG=${CI_COMMIT_REF_SLUG:-master}
+    - export COMMIT_TAG=${DOCKER_TAG}"_"${CI_COMMIT_SHA}
+    - export SERVICE_IMAGE=${DOCKER_REPOSITORY}:${COMMIT_TAG}
```
Comment (lluiscampos) [inline .gitlab-ci.yml:27]: As we are including the template, you can just do:
```suggestion
  before_script:
    - *export_docker_vars
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# integration test runner bot
```
Comment (lluiscampos) [inline README.md:1]: Thanks for adding the README!

File: README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+## Infrastructure
+It's currently hosted on `company-websites` GKE Kubernetes cluster. The cluster in still in POC phase, so, CD is simplified to not have a lot of time vasted if the POC won't go live.
```
Comment (lluiscampos) [inline README.md:4]:
```suggestion
It's currently hosted on `company-websites` GKE Kubernetes cluster. The cluster in still in POC phase, so, CD is simplified to not have a lot of time wasted if the POC won't go live.
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+## Continuous Delivery
+
+### Setup access to GKE
+1. create service account with the following roles assigned: `Kubernetes Engine Developer`, `Kubernetes Engine Service Agent` and `Viewer`
+2. create json key and make base64 encoded hash with removing new lines: `base64 /path/to/saved-key.json | tr -d \\n`
+3. in CI/CD project settings add `GCLOUD_SERVICE_KEY` variable where value is the hash
```
Comment (lluiscampos) [inline README.md:11]: Okey for now, to be moved to a more generic place after the PoC

File: README.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+```
+And deploy correct image manually:
+```
+kubectl set image deployment/test-runner-mender-io test-runner-mender-io=$SERVICE_IMAGE
+```
+
+## Hints
+Get previous revision:
+```
+PREV_REVISION=$(kubectl rollout history deployment test-runner-mender-io | grep ^[0-9] | awk {'print $1'} | tail -n2 | head -n1)
```
Comment (lluiscampos) [inline README.md:30]: :+1:

**Comment by lluiscampos** [comment]
> (...) The only thing is missing is updating k8s deployment yaml in `sre-tools` repo. @lluiscampos can you please advice how to do this?

As discussed, we can leave this as-is for the PoC time and come back later once we have a decision.

For your reference, here are the parts on how it works today.
* A repo merges into `staging` (the deploy branch) runs this job: [trigger:saas:sync-staging-component](https://github.com/mendersoftware/mendertesting/blob/master/.gitlab-ci-check-docker-build.yml#L167)
* Which runs a job inside saas repo to make the code change and commit: [sync:staging-component](https://github.com/mendersoftware/saas/blob/master/.gitlab-ci.yml#L55)
* After that commit, a new pipeline is triggered with this job (the job executes on changes to yaml files): [deploy:staging:services](https://github.com/mendersoftware/saas/blob/master/.gitlab-ci.yml#L110)

File: README.md (hunk 2 above)
Comment (0lmi) [reply README.md:1]: 👍

File: README.md (hunk 4 above)
Comment (0lmi) [reply README.md:11]: Good point. Also thought about it 👌

File: .gitlab-ci.yml (hunk 1 above)
Comment (0lmi) [reply .gitlab-ci.yml:27]: Great! Thanks!

**Pull request merged**